controllers/add_process_groups.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * add_process_groups.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import (
+	ctx "context"
+	"fmt"
+	"time"
+
+	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
+)
+
+// AddProcessGroups provides a reconciliation step for adding new pods to a cluster.
+type AddProcessGroups struct{}
+
+// Reconcile runs the reconciler's work.
+func (a AddProcessGroups) Reconcile(r *FoundationDBClusterReconciler, context ctx.Context, cluster *fdbtypes.FoundationDBCluster) (bool, error) {
+	desiredCountStruct, err := cluster.GetProcessCountsWithDefaults()
+	if err != nil {
+		return false, err
+	}
+	desiredCounts := desiredCountStruct.Map()
+
+	processCounts := make(map[string]int)
+	processGroupIDs := make(map[string]map[int]bool)
```

Comment:
```suggestion
	processGroupIDs := make(map[string]bool)
```

Using a map as a set should be fine if we use the `processGroupID` (which already contains the processClass and the the processGroupID).

Reply:
The problem is that the processGroupID contains other things too, specifically the instance ID prefix, and the instance ID prefix is not contained in the pod name. This means that using the process group ID as the key would cause this method to miss duplicate pod names when changing the instance ID prefix.

Reply:
Maybe that suggests we should be using the pod name in this map instead.